### PR TITLE
feat: conditionally include files

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
In the multifile editor, the contents of `style.css` or `script.js` now only get included if the `index.html` file specifically adds them via a `link` or `script` element.

- [x] create a ~~getOriginalFilename~~ wasHtmlFile function 
- [x] rewrite findIndexHtml to use ~~getOriginalFilename~~ wasHtmlFile

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
If we merge https://github.com/freeCodeCamp/freeCodeCamp/pull/42195 I'll update this to use iframes as needed.

File: client/src/templates/Challenges/rechallenge/builders.js [hunk 1]
```diff
@@ -56,14 +56,16 @@ export const cssToHtml = cond([
...
-// TODO(oliver): relying on includes is quite brittle. If we can keep the
-// original filename in the file or rely on history, that should be safer.
 export function findIndexHtml(files) {
-  const filtered = files.filter(({ includes }) => includes);
+  const filtered = files.filter(file => wasHtmlFile(file));
   if (filtered.length > 1) {
     throw new Error('Too many html blocks in the challenge seed');
   }
-  return filtered.length ? filtered[0] : null;
+  return filtered[0];
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/rechallenge/builders.js:64]: What about when it is `undefined`? The tests still expect `null` to be returned

File: client/src/templates/Challenges/rechallenge/builders.js [hunk 2]
```diff
@@ -89,10 +91,7 @@ A required file can not have both a src and a link: src = ${src}, link = ${link}
 
   const source = files.reduce((source, file) => {
     if (!indexHtml) return source.concat(file.contents, htmlCatch);
-    if (
-      indexHtml.includes.includes(file.history[0]) ||
-      file.history[0] === 'index.html'
-    ) {
+    if (indexHtml.includes.includes(file.history[0]) || wasHtmlFile(file)) {
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/rechallenge/builders.js:94]: Any chance we can change this to not _include_ a property name the same as an in-built function?

Comment (ojeytonwilliams) [reply client/src/templates/Challenges/rechallenge/builders.js:94]: Sure. How about importedFiles?

Comment (ShaunSHamilton) [reply client/src/templates/Challenges/rechallenge/builders.js]: Is it not just a duplicate of `files[<file>].path`? If "no", then that should work.

Comment (ojeytonwilliams) [reply client/src/templates/Challenges/rechallenge/builders.js]: Nope, it's an array containing `index.css` and/or `index.js`, which the transformer needs in order to include or exclude those files from the preview and tests.

**Review by ShaunSHamilton — DISMISSED** [review]
This LGTM 👍 

---

I did come across this error, after submitting a next challenge:
```console
Uncaught Error: Cannot read property 'outputZoneHeight' of null

TypeError: Cannot read property 'outputZoneHeight' of null
    at Editor.getOutputZoneTop (Editor.js:603)
    at Object.getPosition (Editor.js:330)
    at StandaloneEditor.layoutOverlayWidget (codeEditorWidget.js:958)
```

However, I am struggling to reproduce it. 🤷‍♂️

**Review by moT01 — APPROVED** [review]
This all looks to be working, I was able to only see CSS and JS changes after adding a link in the HTML. I tested out a number of existing challenges, none of those appeared broken. The tests are passing, so I assume everything still works - but is there any reason existing challenges might break from this? I assume not, just want to double check. If not, I would be fine merging it.

File: client/src/templates/Challenges/rechallenge/transformers.js [hunk 3]
```diff
@@ -213,6 +214,35 @@ const transformHtml = async function (file) {
...
+    div.querySelector('script[src="script.js"]') ??
+    div.querySelector('script[src="./script.js"]');
+  const importedFiles = [];
+  if (link) {
+    importedFiles.push('index.css');
+    link.remove();
+  }
+  if (script) {
+    importedFiles.push('index.js');
+    script.remove();
```
Comment (nhcarrigan) [inline client/src/templates/Challenges/rechallenge/transformers.js:237]: Removing these means I won't be able to test for them in the document.querySelector results, right?

Comment (ojeytonwilliams) [reply client/src/templates/Challenges/rechallenge/transformers.js:237]: Right. They were causing 404 errors to appear in the console, so I figured it was simpler just to remove them entirely.

**Comment by ojeytonwilliams** [comment]
> This all looks to be working, I was able to only see CSS and JS changes after adding a link in the HTML. I tested out a number of existing challenges, none of those appeared broken. The tests are passing, so I assume everything still works - but is there any reason existing challenges might break from this? I assume not, just want to double check. If not, I would be fine merging it.

No reason I can think of, no.  Weird things do happen, but unless a challenge references `style.css` or `script.js` nothing should change.

File: client/src/templates/Challenges/rechallenge/transformers.js (hunk 3 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/rechallenge/transformers.js:237]: Having said that, if you need something in a test, we can always add attributes and so on.  You want to keep, say, an empty `<script>` element with a custom `data-fcc="js"` attribute?  Something like that?

Comment (nhcarrigan) [reply client/src/templates/Challenges/rechallenge/transformers.js:237]: Nah, I was actually testing that the link element to link the CSS existed lol

Comment (ojeytonwilliams) [reply client/src/templates/Challenges/rechallenge/transformers.js:237]: Okay, cool. If you can manage without, let's keep it simple and keep the number of things in the frame to a minimum.

**Review by naomi-lgbt — APPROVED** [review]
This looks good to me. Solves the issue with linking CSS, and doesn't seem to break anything.

**Pull request merged**